gym/vector/utils/numpy_utils.py

```diff
@@ -108,29 +110,26 @@ def create_empty_array(space, n=1, fn=np.zeros):
                  ('velocity', array([[0., 0.],
                                      [0., 0.]], dtype=float32))])
     """
-    if isinstance(space, _BaseGymSpaces):
-        return create_empty_array_base(space, n=n, fn=fn)
-    elif isinstance(space, Tuple):
-        return create_empty_array_tuple(space, n=n, fn=fn)
-    elif isinstance(space, Dict):
-        return create_empty_array_dict(space, n=n, fn=fn)
-    elif isinstance(space, Space):
-        return create_empty_array_custom(space, n=n, fn=fn)
-    else:
-        raise ValueError(
-            f"Space of type `{type(space)}` is not a valid `gym.Space` instance."
-        )
+    raise ValueError(
+        f"Space of type `{type(space)}` is not a valid `gym.Space` instance."
+    )
 
 
+@create_empty_array.register(Box)
+@create_empty_array.register(Discrete)
+@create_empty_array.register(MultiDiscrete)
+@create_empty_array.register(MultiBinary)
 def create_empty_array_base(space, n=1, fn=np.zeros):
```

Comment:
Opinion: All the concrete implementations for each type should have a name starting with an underscore, to indicate that they shouldn't be called directly, and also not to "pollute" the namespace (to make it easy to understand which function is to be used).

Reply:
Hi @lebrice, thank you for the review appreciate it! And happy new year :)

I like this suggestion. Looking at the changed in #2328, they haven't renamed their concrete implementations to start with an underscore. Therefore, in the interest of avoiding confusion to future readers of the code I can do one of these two:
1) Either we leave it without an underscore
2) I can make the changes to the concrete implementations of flatten and flatten_space functions in #2328 to start with an underscore, and then do the same for the functions being modified in this PR.

@jkterry1 let me know what works best!

Reply:
@RedTachyon what do you want done here?

Reply:
I'd also prefer underscored names with the same reasoning as @lebrice, overall I don't have much to add beyond that

Reply:
Sounds like option 2 it is then @RushivArora

Reply:
Changes are made, should be good now @jkterry1 @lebrice